backend/src/main/java/gov/cdc/usds/simplereport/service/OrganizationService.java

```diff
@@ -175,16 +173,16 @@ public Facility updateFacility(
         a.setPostalCode(orderingProviderZipCode);
         p.setAddress(a);
 
-        // remove all existing devices
-        for(DeviceType d : facility.getDeviceTypes()) {
-            facility.removeDeviceType(d);
+        for (DeviceSpecimen ds : deviceHolder.getConfiguredDeviceTypes()) {
+            facility.addDeviceSpecimen(ds);
         }
-
-        // add new devices
-        for(DeviceType d : devices) {
-            facility.addDeviceType(d);
+        // remove all existing devices
+        for (DeviceSpecimen ds : facility.getDeviceSpecimens()) {
+            if (!deviceHolder.getConfiguredDeviceTypes().contains(ds)) {
+                facility.removeDeviceSpecimen(ds);
+            }
```

Comment:
That looks like it does the same thing, but at least as I recall the way hibernate manages collections, it doesn't when you get down to the DB level (which is why I changed it to this). The slightly quixotic goal is to prevent hibernate from deleting and recreating the many-to-many entries for this relationship every time anybody adds or removes something, which I believe is what happens when you clear the managed collection and then rebuild it. 

It's possible I'm mis-remembering or out of date, and that this would work fine—if so, though, it would probably make more sense to just have `Facility` expose a `setConfiguredDeviceSpecimens` method, rather than have the logic live in the OrganizationService. (Arguably, it should do that anyway, and then Facility would be in charge of deciding whether it needed this logic or a simple `clear()/addAll()` setup.)

Reply:
With the caveat that I don't feel strong enough about this change to die on this hill:

> The slightly quixotic goal is to prevent hibernate from deleting and recreating the many-to-many entries for this relationship

At risk of asking an obvious question, what's the issue with this? I assume not performance, since this is a relatively infrequent call. Does this botch other DB records elsewhere that FK to `facility_device_specimen`?

Reply:
It's not a performance issue here, although I can justify it along those lines because it's the kind of code that people tend to copy-paste, so leaving poorly-performing examples around unnecessarily is just leaving somebody else to discover the problem later on. Honestly, though, it just annoys me when the ORM code does dumb things that it doesn't need to—hence "quixotic" 😁 .